packages/rum-core/src/domain/rumEventsCollection/view/trackViews.ts

```diff
@@ -41,7 +50,7 @@ export function trackViews(
   lifeCycle: LifeCycle,
   onNewLocation: NewLocationListener = () => undefined
 ) {
-  onNewLocation = wrapOnNewLocation(onNewLocation)
+  onNewLocation = catchErrors(onNewLocation, 'onNewLocation threw an error:')
```

Comment:
I'd be in favor of moving this wrapping closer to when we check the configuration.
It could be nice to consistently perform this kind of wrapping during the same phase of the lifecycle and earlier the better imo.